src/zarr/storage/_common.py

```diff
@@ -312,8 +312,27 @@ async def make_store_path(
             # By only allowing dictionaries, which are in-memory, we know that MemoryStore appropriate.
             store = await MemoryStore.open(store_dict=store_like, read_only=_read_only)
         else:
-            msg = f"Unsupported type for store_like: '{type(store_like).__name__}'"  # type: ignore[unreachable]
-            raise TypeError(msg)
+            try:  # type: ignore[unreachable]
+                import fsspec
+
+                if isinstance(store_like, fsspec.mapping.FSMap):
+                    if path:
+                        raise TypeError(
+                            "'path' was provided but is not used for FSMap store_like objects"
+                        )
+                    if storage_options:
+                        raise TypeError(
+                            "'storage_options was provided but is not used for FSMap store_like objects"
+                        )
+                    store = FsspecStore.from_mapper(store_like, read_only=_read_only)
+                else:
+                    raise (
+                        TypeError(f"Unsupported type for store_like: '{type(store_like).__name__}'")
+                    )
+            except ImportError:
+                raise (
+                    TypeError(f"Unsupported type for store_like: '{type(store_like).__name__}'")
+                ) from None
```

Comment:
This is all ugly code because I don't want to assume that fsspec is installed or import it at the module level. I'll look into a better approach similar to xarray's [module_available](https://github.com/pydata/xarray/blob/2870710f543813f6de891b95397cc19e8623af92/xarray/namedarray/utils.py#L34-L60) code.

Reply:
could we pack this logic into a stand-alone function that basically narrows its input to a `FsspecStore` or errors? and maybe we could have some boolean variables in this module that represent if fsspec is present or not, e.g. something like this at the top of the file: 

```python
try:
  import fsspec
  has_fsspec = True
except ImportError:
  has_fsspec = False
```